recipes/pcapplusplus/cmake/conanfile.py

```diff
@@ -0,0 +1,60 @@
+import os
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake
+from conan.tools.files import copy, get, rmdir
+from conan.errors import ConanInvalidConfiguration
+
+
+required_conan_version = ">=1.57.0"
+
+class PcapplusplusConan(ConanFile):
+    name = "pcapplusplus"
+    package_type = "static-library"
+    license = "Unlicense"
+    description = "PcapPlusPlus is a multiplatform C++ library for capturing, parsing and crafting of network packets"
+    topics = ("pcap", "network", "security", "packet")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/seladb/PcapPlusPlus"
+    settings = "os", "arch", "build_type", "compiler"
+    options = {
+        "immediate_mode": [True, False],
+    }
+    default_options = {
+        "immediate_mode": False,
+    }
+
+    def requirements(self):
+        if self.settings.os == "Windows":
+            self.requires("npcap/1.70")
+        else:
+            self.requires("libpcap/1.10.1")
+
+    def validate(self):
+        if self.settings.os not in ("FreeBSD", "Linux", "Macos", "Windows"):
+            raise ConanInvalidConfiguration(f"{self.settings.os} is not supported")
+
+    def source(self):
+        get(
+            self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True
+        )
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["PCAPPP_BUILD_TESTS"] = False
+        tc.variables["PCAPPP_BUILD_EXAMPLES"] = False
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"), keep_path=False)
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
```

Comment:
```suggestion
    def package_info(self):
        self.cpp_info.set_property("cmake_file_name", "PcapPlusPlus")
        self.cpp_info.set_property("cmake_target_name", "PcapPlusPlus::PcapPlusPlus")
```

I see the upstream generated cmake files are nor lower case:

```
Install the project...
/opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/bin/cmake -P cmake_install.cmake
-- Install configuration: "Release"
-- Installing: /tmp/install/lib/pkgconfig/PcapPlusPlus.pc
-- Installing: /tmp/install/lib/cmake/pcapplusplus/FindPCAP.cmake
-- Installing: /tmp/install/lib/cmake/pcapplusplus/PcapPlusPlusTargets.cmake
-- Installing: /tmp/install/lib/cmake/pcapplusplus/PcapPlusPlusTargets-release.cmake
-- Installing: /tmp/install/lib/cmake/pcapplusplus/PcapPlusPlusConfig.cmake
-- Installing: /tmp/install/lib/cmake/pcapplusplus/PcapPlusPlusConfigVersion.cmake
```


Plus, I found individual targets, like ` PcapPlusPlus::Packet++ PcapPlusPlus::Pcap++ PcapPlusPlus::Common++` that much better. It's possible to add here in cpp_info too as [property](https://docs.conan.io/2/reference/tools/cmake/cmakedeps.html#properties)

Reply:
There is also this `PcapPlusPlus.pc`, so `pkg_config_name` property should be set accordingly.

Reply:
Fixed in Fixed in https://github.com/conan-io/conan-center-index/pull/20296/commits/9f304cbee9a2a61149277c00d741bb16ec521467